### ISSUE TITLE
Fix JSON-RPC `eth_getTransactionCount` method

### ISSUE THREAD

**Issue by hackaugusto** [ethereum/pyethapp#133]
```
    @public
    @decode_arg('address', address_decoder)
    @decode_arg('block_id', block_id_decoder)
    @encode_res(quantity_encoder)
    def getTransactionCount(self, address, block_id='pending'):
        block = self.json_rpc_server.get_block(block_id)
        return block.get_nonce(address)
```

The current implementation returns the `nonce` instead of the transaction count.

**Comment by heikoheiko** [comment]
https://github.com/ethereum/pyethereum/blob/develop/ethereum/blocks.py#L717
https://github.com/ethereum/pyethereum/blob/develop/ethereum/config.py#L45

**Comment by RomanZacharia** [comment]
@konradkonrad @hackaugusto  The nonce does represent the number of transactions sent from a specific address within a block. So seems like returning a nonce is absolutely fine, at least the cpp-ethereum works in the same way https://github.com/ethereum/libethereum/blob/662fb364085e7dd60c0a974d3e4f2e67821dee6f/libethereum/State.cpp#L352

**Comment by heikoheiko** [comment]
No, that's not correct. The nonce might start at a higher number, depending on the configuration. You need to subtract the start nonce from the nonce to get the transaction count. 

You need to factor in the config.eth.ACCOUNT_INITIAL_NONCE or similar. 
https://github.com/ethereum/pyethereum/blob/develop/ethereum/config.py#L45
https://github.com/ethereum/pyethapp/blob/develop/pyethapp/profiles.py#L46

**Comment by konradkonrad** [comment]
also see https://github.com/ethereum/pyethapp/pull/144

**Comment by Smithgift** [comment]
This current behavior is inconsistent with the other implementations, which do return the nonce. See: [here](https://github.com/karalabe/hive/pull/22).

**Comment by konradkonrad** [comment]
> This current behavior is inconsistent with the other implementations, which do return the nonce

yeah, you are correct!
